src/ethereum/EthereumTools.ts

```diff
@@ -205,6 +206,21 @@ export class EthereumTools implements EdgeCurrencyTools {
     }
     address = contractAddress
 
+    // Split the address to get the chainId according to EIP-681
+    if (address.includes('@')) {
```

Comment:
Does this allow using an Ethereum: uri with an L2 like optimism if and only if the chainId exists?